### PR TITLE
chore: remove EoL Node.js 16 from test matrix

### PR DESCRIPTION
**Pull request by rluvaton**

### CODE REVIEW

**Review by RafaelGSS — APPROVED** [review]

**Review by benjamingr — APPROVED** [review]

**Review by VoltrexKeyva — APPROVED** [review]

**Comment by Trott** [comment]
Can you please update the "engines" field in package.json as well?

**Comment by Trott** [comment]
Totally not necessary for this PR but optional here or as a future enhancement: If we use the pkgjs prepare-node-test-matrix GitHub Action instead of hardcoding the Node.js version numbers, it will look at the "engines" property in package.json and only test the LTS versions at or above that level along with the current version. See https://github.com/pkgjs/action/tree/main for more information. (pkgjs is the tooling org managed in a very hands-off way by the TSC for Node.js.)

**Comment by rluvaton** [comment]
@Trott moved to `prepare-node-test-matrix` 😄

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "Utilities for Node.js core collaborators",
   "type": "module",
   "engines": {
-    "node": ">=16.8.0"
+    "node": ">=18.0.0"
```
Comment (aduh95) [inline package.json:7]:
```suggestion
    "node": "^18.18.0 || >=20.0.0"
```

Comment (rluvaton) [reply package.json:7]: why the 20 is needed?

Comment (aduh95) [reply package.json:7]: 20.x is the current release line, it makes sense to support it. My suggestion is to not use a range that includes 19.x (notice the `^` before `18.18`).

Comment (rluvaton) [reply package.json:7]: oh, I forgot 19 is dead

**Pull request merged**